src/pipdeptree/_models/dag.py

```diff
@@ -36,19 +36,14 @@ class PackageDAG(Mapping[DistPackage, List[ReqPackage]]):
     """
 
     @classmethod
-    def from_pkgs(cls, pkgs: list[DistInfoDistribution]) -> PackageDAG:
+    def from_pkgs(cls, pkgs: list[Distribution]) -> PackageDAG:
         dist_pkgs = [DistPackage(p) for p in pkgs]
         idx = {p.key: p for p in dist_pkgs}
         m: dict[DistPackage, list[ReqPackage]] = {}
         for p in dist_pkgs:
             reqs = []
             for r in p.requires():
-                # Requirement key is not sufficiently normalized in pkg_resources - apply additional normalization
-                d = idx.get(pep503_normalize(r.key))
-                # pip's _vendor.packaging.requirements.Requirement uses the exact casing of a dependency's name found in
-                # a project's build config, which is not ideal when rendering.
-                # See https://github.com/tox-dev/pipdeptree/issues/242
-                r.project_name = d.project_name if d is not None else r.project_name
+                d = idx.get(pep503_normalize(r.name))
```

Comment:
Due to what was said [here](https://github.com/tox-dev/pipdeptree/pull/333/files#r1540466505), do the following:

```suggestion
                r.name = d.project_name if d is not None else r.name
                d = idx.get(pep503_normalize(r.name))
```

Let's also add a comment above this line to clarify why we are changing `r.name`

Reply:
Done.

Reply:
Besides this, we also need to remove the `lower()` in `mock_pkgs`.

That's all. Done this part.